sync/class.jetpack-sync-actions.php

```diff
@@ -157,13 +157,19 @@ static function schedule_initial_sync() {
 	}
 
 	static function schedule_full_sync( $modules = null ) {
+		if ( ! self::sync_allowed() ) {
+			return false;
```

Comment:
We should consider returning errors from the API instead of `false`. Then in Calypso (or other UIs), we can properly respond to different errors.